src/core/crypto/hmac_sha256.hpp

```diff
@@ -87,13 +90,37 @@ class HmacSha256
     void Start(const uint8_t *aKey, uint16_t aKeyLength);
 
     /**
-     * This method inputs bytes into the HMAC computation.
+     * This method feeds bytes into the HMAC computation.
      *
      * @param[in]  aBuf        A pointer to the input buffer.
      * @param[in]  aBufLength  The length of @p aBuf in bytes.
      *
      */
-    void Update(const uint8_t *aBuf, uint16_t aBufLength);
+    void FeedBytes(const void *aBuf, uint16_t aBufLength);
```

Comment:
nit: most hash APIs I've seen use "update" or "input". Is there precedent for using "feed"?

Reply:
Agree with @jwhui. If there is no semantic changes between `update` and `feed`, I would prefer keeping `update`.

Reply:
I wanted  to somehow include the `Message` in the name of the method which reads the data from a `Message`. Initially I had it called `UpdateMessage()` which felt weird to me (it is as if we are updating the message and not reading from it)... Also considered `UpdateFromMessage()` but I feel it's too long.  So I changed them all to `Feed()`  and used `FeedMessage()`...  😄 

I don't have any strong preference here. If `Update` is commonly used I am ok using  `Update` for this. I will then overload the method name `Update()` for all flavors . We can tell them apart from the number and types  of the passed arguments - which I think is fine. I will go with this for now.

Thoughts/suggestions?

Reply:
Overloading the method works for me.